### PR TITLE
Feature/sld 30

### PR DESCRIPTION
**Pull request by brandon9912**
### Description:

- Notion link: https://www.notion.so/SLD-30-Client-Integrate-Client-API-465b83f32346475ea4620e6d5ededebe?pvs=4
- add start lucky draw API

### Changes:
- integrate 4 client API

### CODE REVIEW

File: server/controllers/eventsController.js [hunk 1]
```diff
@@ -10,7 +10,133 @@ const agent = new https.Agent({
...
+          // 拿取會員近一個月消費金額
+          axios
+            .get(`http://order-api.appworks.local/api/v1/admin/orders`, {
+              httpsAgent: agent,
+              params: { member_id: member_id },
+            })
+            .then(async (response) => {
+              // timestamp of one month age
+              const oneMonthAgo =
+                new Date().getTime() - 30 * 24 * 60 * 60 * 1000;
```
Comment (YuakaTL) [inline server/controllers/eventsController.js]: Nit: You can use dayjs to format the date and time, which will make it more understandable.

File: server/controllers/eventsController.js [hunk 2]
```diff
@@ -10,7 +10,133 @@ const agent = new https.Agent({
...
+                (sum, order) => sum + order.order_amount,
+                0
+              );
+              console.log(
+                `近一個月內的金額總和為 ${totalAmount - totalRefundAmount} 元。`
+              );
+              const memberAmount = totalAmount - totalRefundAmount;
+              // 用GET /event 拿取獎品列表，使用最低 threshold 檢查是否符合抽獎資格，若庫存為 0 也回傳不能抽獎
+              axios
+                .get(`http://localhost:5000/api/v1/lottery/event`, {
```
Comment (YuakaTL) [inline server/controllers/eventsController.js]: suggest use env to config API URL
Like:
```
API_DEV=http://development:3000
API_PRO=http://production

process.env.NODE_ENV === 'development' ? process.env.API_DEV : process.env.API_PRO
```

File: server/controllers/eventsController.js [hunk 3]
```diff
@@ -10,7 +10,133 @@ const agent = new https.Agent({
...
+
+                    discount_inventory = discount.inventory;
+                    // discount_threshold = discount.threshold
+                  }
+                  // 用 PUT /inventory 將庫存-1
+                  console.log(discount);
+                  console.log(discount.discount_id);
+                  axios
+                    .put(
+                      `http://localhost:5000/api/v1/lottery/inventory/${discount.discount_id}`,
```
Comment (YuakaTL) [inline server/controllers/eventsController.js]: ditto

Comment (egnkvn) [inline server/controllers/eventsController.js]: 這邊是不是要換成http://192.168.50.104/

**Review by YuakaTL — DISMISSED** [review]

File: server/controllers/eventsController.js (hunk 2 above)
Comment (egnkvn) [inline server/controllers/eventsController.js:79]: 這邊是不是要換成http://192.168.50.104/

**Review by 5-michelle — APPROVED** [review]
LGTM!

**Pull request merged**